Source/WebCore/layout/formattingContexts/inline/InlineLineBuilder.cpp

```diff
@@ -1068,6 +1068,46 @@ LayoutUnit LineBuilder::adjustGeometryForInitialLetterIfNeeded(const Box& floatB
     return initialLetterCapHeightOffset.value_or(0_lu);
 }
 
+bool LineBuilder::placedFloatsContainsLeftPositionedFloat() const
+{
+    ASSERT(floatingState());
+    auto floatingContext = FloatingContext { formattingContext(), *floatingState() };
+    for (auto floatItem : m_placedFloats) {
+        if (floatingContext.toFloatItem(floatItem->layoutBox()).isLeftPositioned())
+            return true;
+    }
+    return false;
+}
+
+bool LineBuilder::placedFloatsContainsRightPositionedFloat() const
+{
+    ASSERT(floatingState());
+    auto floatingContext = FloatingContext { formattingContext(), *floatingState() };
+    for (auto floatItem : m_placedFloats) {
+        if (floatingContext.toFloatItem(floatItem->layoutBox()).isRightPositioned())
+            return true;
+    }
+    return false;
+
+}
+
+void LineBuilder::trimMarginFromFloatGeometry(BoxGeometry& floatGeometry, MarginTrimType marginTrimType)
+{
+    switch (marginTrimType) {
+    case MarginTrimType::InlineStart:
+        floatGeometry.setHorizontalMargin({ 0_lu, floatGeometry.horizontalMargin().end });
+        return;
+    case MarginTrimType::InlineEnd:
+        floatGeometry.setHorizontalMargin({ floatGeometry.horizontalMargin().start, 0_lu });
+        return;
+    case MarginTrimType::BlockStart:
+        floatGeometry.setVerticalMargin({ 0_lu, floatGeometry.verticalMargin().after });
+        return;
+    default:
+        ASSERT_NOT_REACHED();
+    }
+}
```

Comment:
this looks like a simple helper, please turn this into a static function...also I am not sure if the naming is quite correct as there is no assertion that this always trims float box. I'd just call it trimMarginFromBox. Actually this could be a static function on InlineFormattingGeometry with that generic name.